tests/sles4sap/netweaver_cluster.pm

```diff
@@ -0,0 +1,126 @@
+
+    # Set SAP variables
+    my $pscmd = $self->set_ps_cmd("$type");
+    $self->set_sap_info($useradm);
+    $self->set_sap_instance($instance_id);
+
+    # Synchronize the nodes
+    barrier_wait "NW_CLUSTER_INSTALL_$cluster_name";
+
+    select_console 'root-console';
+
+    # Do stuff with SAP account
+    $self->become_sapadm;
+    $self->test_version_info;
+    $self->test_instance_properties;
+    $self->test_stop;
+    script_run "$pscmd | wc -l ; $pscmd";
+    save_screenshot;
+
+    # Rollback changes to $testapi::serialdev and close the window
+    type_string "exit\n";
+    ensure_serialdev_permissions;
+
+    # Some file changes are needed for HA
+    select_console 'root-console';
+
+    my $profile_file = "/usr/sap/$sid/SYS/profile/${sid}_${type}${instance_id}_${alias}";
+
+    assert_script_run 'cp /sapinst/sapservices /usr/sap/';
+    assert_script_run "echo 'service/halib = \${DIR_CT_RUN}/saphascriptco.so' >> $profile_file";
+    assert_script_run "echo 'service/halib_cluster_connector = /usr/bin/sap_suse_cluster_connector' >> $profile_file";
+    assert_script_run "sed -i 's/^Restart_Program_\\(.*[[:blank:]]*=\\)/Start_Program_\\1/' $profile_file";
+
+    # Add SAP account into haclient group
+    assert_script_run "usermod -a -G haclient $useradm";
+
+    # Removed uneeded(?) profile directory for ERS
+    if ($type eq 'ERS') {
+        assert_script_run "rm -rf /usr/sap/$sid/${type}${instance_id}/profile";
+        assert_script_run "ln -s /sapmnt/$sid/profile /usr/sap/$sid/${type}${instance_id}/profile";
+        assert_script_run "chown -h ha1adm:sapsys /usr/sap/$sid/${type}${instance_id}/profile";
+    }
+
+    # Create the resource configuration
+    my @sedoptions = undef;
+    push @sedoptions, "-e 's|%SID%|$sid|g'";
+    push @sedoptions, "-e 's|%${type}_INSTANCE%|$instance_id|g'";
+    push @sedoptions, "-e 's|%${type}_ALIAS%|$alias|g'";
+    push @sedoptions, "-e 's|%${type}_LUN%|$lun|g'";
+    push @sedoptions, "-e 's|%${type}_IP%|$ip|g'";
+
+    # We need to execute the sed command node by node
+    my $cmd             = undef;
+    my $nw_cluster_conf = '/sapmnt/nw_cluster.conf';
+    if (is_node(1)) {
+        # Initiate the template
+        $cmd = 'sed' . join(' ', @sedoptions) . " /sapinst/nw_cluster.conf > $nw_cluster_conf";
+        assert_script_run $cmd;
+
+        # Synchronize the nodes
+        barrier_wait "NW_INIT_CONF_$cluster_name";
+        barrier_wait "NW_CREATED_CONF_$cluster_name";
+
+        # Upload the configuration into the cluster
+        assert_script_run 'crm configure property maintenance-mode=true';
+        assert_script_run "crm configure load update $nw_cluster_conf";
+        assert_script_run 'crm configure property maintenance-mode=false';
+    }
+    else {
+        # Synchronize the nodes
+        barrier_wait "NW_INIT_CONF_$cluster_name";
+
+        # Use mutex to be sure that only *one* node at a time can access the file
+        mutex_lock 'support_server_ready';
```

Comment:
Yes, it's not the same usage as `barrier_wait`. Mutex is here to lock a resource and can be used multiple time, but barrier can be used only once and is used to synchronize nodes.

Note: before you ask, I'm using `support_server_ready` mutex because it is already created and I don't want to create a new one just for this (and mutexes should be created in the support_server, I was never be able to create them outside of the support_server).